src/commands/dependencies/__tests__/upload.test.ts

```diff
@@ -0,0 +1,405 @@
+    expect(stdout[2]).toEqual(`${chalk.bold('Service')}: my-service`)
+    expect(stdout[3]).toEqual(`${chalk.bold('Version')}: 1.234`)
+    expect(stdout[4]).toEqual('')
+    expect(stdout[5]).toEqual('Uploading dependencies...')
+    expect(stdout[6]).toEqual(chalk.red('Failed upload dependencies: No access granted'))
+
+    expect(code).toBe(3)
+  })
+
+  test('handles API 403 errors', async () => {
+    ;(axios.post as jest.Mock).mockImplementation(() =>
+      Promise.reject({message: 'Forbidden', isAxiosError: true, response: {status: 403}})
+    )
+
+    const {context, code} = await runUploadDependenciesCommand(
+      './src/commands/dependencies/__tests__/fixtures/dependencies',
+      {
+        apiKey: 'DD_API_KEY_EXAMPLE',
+        appKey: 'DD_APP_KEY_EXAMPLE',
+        releaseVersion: '1.234',
+        service: 'my-service',
+        source: 'snyk',
+      }
+    )
+
+    const stdout = context.getStdoutBuffer().split(os.EOL)
+    const stderr = context.getStderrBuffer().split(os.EOL)
+
+    expect(stderr).toEqual(['Forbidden'])
+    expect(stdout[6]).toEqual(
+      chalk.red(
+        `Failed upload dependencies: Forbidden. Check ${chalk.bold('DATADOG_API_KEY')} and ${chalk.bold(
+          'DATADOG_APP_KEY'
+        )} environment variables.`
+      )
+    )
+
+    expect(code).toBe(3)
+  })
+})
+
+interface RunUploadDependenciesInput {
+  apiKey?: string
+  appKey?: string
+  dryRun?: boolean
+  releaseVersion?: string
+  service?: string
+  source?: string
+}
+
+async function runUploadDependenciesCommand(filePath: string, input: RunUploadDependenciesInput) {
+  const cli = new Cli()
+  cli.register(UploadCommand)
+
+  const context = createMockContext()
+
+  process.env = {
+    DATADOG_API_KEY: input.apiKey,
+    DATADOG_APP_KEY: input.appKey,
+  }
+  const params = ['dependencies', 'upload', filePath]
+
+  if (input.releaseVersion) {
+    params.push('--release-version', input.releaseVersion)
+  }
+  if (input.service) {
+    params.push('--service', input.service)
+  }
+  if (input.source) {
+    params.push('--source', input.source)
+  }
+  if (input.dryRun) {
+    params.push('--dry-run')
+  }
+
+  const code = await cli.run(params, context)
+
+  return {context, code}
+}
+
+interface MockContext extends BaseContext {
+  getStderrBuffer(): string
+  getStdoutBuffer(): string
+}
+
+const createMockContext = (): MockContext => {
+  const stdoutChunks: any[] = []
+  const stderrChunks: any[] = []
+
+  return {
+    getStderrBuffer: () => stderrChunks.join(''),
+    getStdoutBuffer: () => stdoutChunks.join(''),
+    stderr: new Writable({
+      write(chunk, encoding, callback) {
+        stderrChunks.push(chunk)
+        callback()
+      },
+    }),
+    stdin: new Readable(),
+    stdout: new Writable({
+      write(chunk, encoding, callback) {
+        stdoutChunks.push(chunk)
+        callback()
+      },
+    }),
+  }
+}
+
+function streamToString(stream: Readable): Promise<string> {
```

Comment:
Why using both format of functions? 
(`const a = () => {}` and `function b() {}`)

Reply:
I have no preference - maybe we could have a tslint rule to ban named functions if you want to keep consistency? :)

Reply:
Definitely, I thought we already had. I'll push a PR for it. No need to update your PR.

Reply:
Oh, I didn't notice your comment - I added a tslint rule in this PR

Reply:
That's ok, I was afraid of the diff it would produce. 
But it's manageable after all.